### PR TITLE
feat: Separate Private and Public Speaker and Session Info

### LINKED ISSUE #6222: Separate Private and Public Speaker and Session Info into two separate pages for Organizers/Speakers vs. Public
The current implementation of the speakers and session info is confusing. As an organizer it is unclear if information is public or if the organizer is viewing personal speaker or session information. 

The user interface of public pages should be incorporated into the public event pages. Private pages are part of the organizer interface. Therefore the content areas should be clearly separated as follows:

1. There should be no edit or status change option on public pages.

2. **Session Pages** - The session pages should have the information in this URL structure
* Public for attendees (with only public info):
/e/[event-number]/session/[session-number]
* Private - Access for speakers and organizer (with complete info)
/events/[event-number]/sessions/[session-number]/
/events/[event-number]/sessions/[session-number]/edit

3. **Speaker Pages** - The speaker pages should have the information in this URL structure
* Public for attendees (with only public info): 
/e/[event-number]/speaker/[session-number]
* Private - Access for speakers and organizer (with complete info)
/events/[event-number]/speaker/[speaker-number]

### PR DESCRIPTION
**Pull request by sachinchauhan2889**
Fixes #6222 

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by sachinchauhan2889** [comment]
@iamareebjamal sir, i am stucking at a place. 
error

sessions are not loading onto view speaker page.
it tried to hard and apend too time but not able to solve.
can you look into this.

**Comment by iamareebjamal** [comment]
Will look in the morning

**Comment by sachinchauhan2889** [comment]
> Will look in the morning

solved sir

**Comment by sachinchauhan2889** [comment]
please check basic structure and routes.
display of only public info in public pages and display of all info in events page is still remaining. i am working on this.

**Comment by mariobehling** [comment]
Thanks!

* The edit URL of speaker and sessions pages is following a different pattern compared to the "view" page
* Please keep the "View" pages very plain similar to other areas of the organizer UI.

Screenshot from 2021-01-04 23-32-13

**Comment by sachinchauhan2889** [comment]
> Please keep the "View" pages very plain and similar to other areas of the organizer UI.

@mariobehling sir, do you want view page similar to edit page. Instead of input box, do we want value of that field in "p" tag

**Comment by mariobehling** [comment]
Yes, please keep it simple and similar to the edit page.

**Comment by mariobehling** [comment]
How can we get this issue finished before you take up new issues?

**Comment by sachinchauhan2889** [comment]
@mariobehling sir, Is it suitable. Please suggest changes if any

feedback

**Comment by mariobehling** [comment]
Thank you. Please see the following things that need to be changed on the session page:

- Add session state button above form following colors of states (pending, accepted etc.)
- Show "Edit" button above form
- Show "Withdraw" button above form
- No space in front of ":"
- URL in capital
- view Session -> View Session
- Exactly follow order in the form
- Test custom form on the bottom
- Do not show html code, but convert it into html
- Do not add a comma after a speaker name, if there is only one speaker and add a comma if there are two speakers
- Test audio and video fields
- Add URLs, e.g. Twitter and test it

**Comment by mariobehling** [comment]
- Not sure where "Timing" comes from. Is it the scheduled time of the session? If yes, please add it above the form in a box "Scheduled Date and Time: ...". If not yet scheduled show "Not yet scheduled"
- For the session type there is a second field for the time. Please add the time info e.g. 00.30h here as well. This would be taken from the wizard data
Screenshot from 2021-01-13 12-13-06

**Comment by sachinchauhan2889** [comment]
Ok sir. I am implementing your suggested changes. Thank you.

**Comment by mariobehling** [comment]
Please also add a red star * if an item has been made "required" by the organizer

**Comment by sachinchauhan2889** [comment]
work is still in progress. comma between speakers and sessions is remaining. Review other parts

**Comment by mariobehling** [comment]
Thank you! Good progress.

* Timing shows current time still
* Form does not show duration of sessions yet
* Please also show form fields if speaker did not fill it in, e.g. if speaker did not fill in "Comments" just show "Comments" on the left and nothing on the right.
* After editing form, return to "View Session" instead of returning to session overview.

**Comment by sachinchauhan2889** [comment]
> Thank you! Good progress.
> 
> * Timing shows current time still
> * Form does not show duration of sessions yet
> * Please also show form fields if speaker did not fill it in, e.g. if speaker did not fill in "Comments" just show "Comments" on the left and nothing on the right.
> * After editing form, return to "View Session" instead of returning to session overview.


thank you sir for review. working on comma between speakers, preventing grid from breaking, and above changes.

**Comment by mariobehling** [comment]
Question: Where could be displayed if a field is public or not? Maybe have a public badge for each form field?

**Comment by sachinchauhan2889** [comment]
> Question: Where could be displayed if a field is public or not? Maybe have a public badge for each form field?


ok sir, i will also try to implement a public badge for every public field.

**Comment by sachinchauhan2889** [comment]
PR comment pic

**Comment by mariobehling** [comment]
Thanks. Good work.

Please change the following:
* The withdraw feature does not work as expected. Maybe it is better to solve this in a separate issue. Could you take it out please for now.

**Comment by mariobehling** [comment]
On the speakers view page:

* There is a broken image on the top left, that should not be there
* Please show the speaker image on the view page instead of the API link
* Do not show a colon behind "Sessions:"
Screenshot from 2021-01-14 20-29-35

**Comment by mariobehling** [comment]
Please change
* Do not show a colon on the edit session page for "Speakers:"
* On heading of speakers page change speaker details -> Speaker details
* Use the heading area of the "Edit Speaker" page as an example and make the other view/edit speaker and session pages the same with the horizontal line

Screenshot from 2021-01-14 20-32-12

File: app/templates/components/events/view/overview/speaker-view.hbs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <div class="ui grid">
     {{#each @fields as |field|}}
-        {{#if field.isIncluded}}
+        {{#if (and field.isIncluded (not-eq field.fieldIdentifier 'photoUrl'))}}
```
Comment (iamareebjamal) [inline app/templates/components/events/view/overview/speaker-view.hbs:3]: Shouldn't this be field.type image?

**Comment by mariobehling** [comment]
Thanks, looks like you have taken care of all the points except one. Please change:
* on the speakers view page, the image of speakers should be in the same area as on the edit page and in the same size.

Screenshot from 2021-01-14 22-54-08

**Comment by mariobehling** [comment]
Please also ensure that all fields are translatable including the Public/Private badges. Thanks.

**Comment by sachinchauhan2889** [comment]
new pr pic

**Review by mariobehling — APPROVED** [review]
Looks good to me. Thank you!
I will follow up in further issues, e.g. we need to clarify how the lock/unlock feature should ultimately work for different roles.

**Comment by iamareebjamal** [comment]
Great Work!

**Pull request merged**